File: subprojects/java-compatibility/src/test/java/kotools/types/internal/KotoolsTypesPackageTest.java
/*
 * Copyright 2023 Kotools S.A.S.U.
 * Use of this source code is governed by the MIT license.
 */

package kotools.types.internal;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class KotoolsTypesPackageTest {
    @Test
    public void toString_should_pass_on_Number() {
        final String actual = KotoolsTypesPackage.Number.toString();
        final String expected = "kotools.types.number";
        Assertions.assertEquals(expected, actual);
    }

    @Test
    public void toString_should_pass_on_Text() {
        final String actual = KotoolsTypesPackage.Text.toString();
        final String expected = "kotools.types.text";
        Assertions.assertEquals(expected, actual);
    }
}

File: subprojects/internal/src/commonTest/kotlin/kotools/types/internal/KotoolsTypesPackageTest.kt
package kotools.types.internal

import kotlin.test.Test
import kotlin.test.assertEquals

@InternalKotoolsTypesApi
class KotoolsTypesPackageTest {
    @Test
    fun toString_should_pass_on_each_values(): Unit = KotoolsTypesPackage
        .entries
        .map { "kotools.types.${it.name.lowercase()}" to "$it" }
        .forEach { assertEquals(expected = it.first, actual = it.second) }
}
